examples/md-flexible/src/Simulation.cpp

```diff
@@ -631,6 +643,62 @@ void Simulation::checkNumParticles(size_t expectedNumParticlesGlobal, size_t num
   }
 }
 
+void Simulation::loadParticles() {
+  // When loading checkpoints, the config file might contain particles that do not belong to this rank,
+  // because rank boundaries don't align with those at the end of the previous simulation due to dynamic load balancing.
+  // Idea: Only load what belongs here and send the rest away.
+  _autoPasContainer->addParticlesIf(_configuration.particles, [&](auto &p) {
+    if (_domainDecomposition->isInsideLocalDomain(p.getR())) {
+      // Mark particle in vector as dummy, so we know it has been inserted.
+      p.setOwnershipState(autopas::OwnershipState::dummy);
+      return true;
+    }
+    return false;
+  });
+
+  // Remove what has been inserted. Everything that remains does not belong into this rank.
+  _configuration.particles.erase(std::remove_if(_configuration.particles.begin(), _configuration.particles.end(),
+                                                [&](const auto &p) { return p.isDummy(); }),
+                                 _configuration.particles.end());
+
+  // Send all remaining particles to all ranks
+  // TODO: This is not optimal but since this only happens once upon initialization it is not too bad.
+  //       Nevertheless it could be improved by determining which particle has to go to which rank.
+  const auto rank = _domainDecomposition->getDomainIndex();
+  ParticleCommunicator particleCommunicator(_domainDecomposition->getCommunicator());
+  for (int receiverRank = 0; receiverRank < _domainDecomposition->getSubdomainCount(); ++receiverRank) {
+    // don't send to ourselves
+    if (receiverRank == rank) {
+      continue;
+    }
+    particleCommunicator.sendParticles(_configuration.particles, receiverRank);
+  }
+  // Erase all locally stored particles. They don't belong to this rank and have been sent away.
+  _configuration.flushParticles();
+
+  // Receive particles from all other ranks.
+  for (int senderRank = 0; senderRank < _domainDecomposition->getSubdomainCount(); ++senderRank) {
+    // don't send to ourselves
+    if (senderRank == rank) {
+      continue;
+    }
+    particleCommunicator.receiveParticles(_configuration.particles, senderRank);
+  }
+  particleCommunicator.waitForSendRequests();
+
+  // Remove duplicates. This is a safety mechanism for the odd case that two ranks loaded the same particle.
```

Comment:
What might be a reason for this to occur?

Furthermore, this appears to only protect against a duplicate particle coming from 2 other ranks but not if the duplicate comes from this rank and one other.

Reply:
> What might be a reason for this to occur?

Honestly no idea. I don't think this can happen, hence I removed this check -> bfc0bdf2d